### PR TITLE
Cargo clippy recommendations & spelling fixes

### PR DESCRIPTION
**Pull request by dralley**
Follow a couple of the recommendations that "cargo clippy" provides, made a couple small ones of my own, & fixed a couple of misspellings.

### CODE REVIEW

File: src/draw_target.rs [hunk 1]
```diff
@@ -390,26 +390,22 @@ impl DrawTarget {
...
     fn line_to(&mut self, pt: Point) {
-        if self.current_point.is_none() {
-            self.current_point = Some(pt);
-            self.first_point = Some(pt);
-        }
         if let Some(current_point) = self.current_point {
             self.rasterizer
                 .add_edge(current_point, pt, false, Point::new(0., 0.));
             self.current_point = Some(pt);
+        } else {
```
Comment (dralley) [inline src/draw_target.rs:397]: Clippy didn't suggest this, but it makes sense (I think)

File: src/geom.rs [hunk 2]
```diff
@@ -10,13 +10,6 @@ pub fn intrect<T: Copy>(x1: T, y1: T, x2: T, y2: T) -> euclid::default::Box2D<T>
     euclid::default::Box2D::new(euclid::point2(x1, y1), euclid::point2(x2, y2))
 }
 
-pub fn abs(a: f32) -> f32 {
-    if a < 0. {
-        return -a;
-    }
-    return a;
-}
-
```
Comment (dralley) [inline src/geom.rs:19]: ```.abs()``` is implemented for all the basic types

File: src/draw_target.rs (hunk 1 above)
Comment (dralley) [reply src/draw_target.rs:397]: Do you mean, using ```move_to```, or the ```if let {} else {}```, or both?

Comment (dralley) [reply src/draw_target.rs]: I went ahead and deleted those changes entirely.

**Comment by jrmuizel** [comment]
Thanks.

File: src/draw_target.rs (hunk 1 above)
Comment (jrmuizel) [reply src/draw_target.rs]: for quad_to and cubic_to we want to still add the rest of the curve and not just do a move_to

**Pull request merged**